test/RToken.test.ts

```diff
@@ -1338,13 +1344,13 @@ describe(`RTokenP${IMPLEMENTATION} contract`, () => {
       })
     })
 
-    it('Should not allow melt to set BU exchange rate to below 1e-9', async () => {
-      // melt()
-      await expect(
-        rToken.connect(addr1).melt(issueAmount.sub(issueAmount.div(bn('1e9'))).add(1))
-      ).to.be.revertedWith('BU rate out of range')
-      await rToken.connect(addr1).melt(issueAmount.sub(issueAmount.div(bn('1e9'))))
-    })
+    // it('Should not allow melt to set BU exchange rate to below 1e-9', async () => {
```

Comment:
We should keep this test, just need to do a larger initial mint for it

Reply:
i don't think its possible to reach that error when melting via the new code (the new require catches first)

Reply:
The new require throws when the remaining balance after melting is < FIX_ONE. We can still get to the 1e9 and 1e-9 exchange rates, we just need to start with a much higher totalSupply. I think if supply is 1e28 we can set basketsNeeded to 1e19-1 and 1e37+1.